File: 普天二代证开发包2011/BCB/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
#include "cardapi.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::BtnOpenClick(TObject *Sender)
{
        TCHAR szErrorText[32];

        OpenCardReader(0,2,115200);
        GetErrorText(szErrorText,32);
        EditResult->Text=szErrorText;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::BtnReadClick(TObject *Sender)
{
        long result;
        PERSONINFO person;
        TCHAR szErrorText[32];

        result=GetPersonMsg(&person,TEXT("ssss.bmp"));
        if(0==result)
        {
                Edit1->Text=person.name;
                Edit2->Text=person.sex;
                Edit3->Text=person.nation;
                Edit4->Text=person.birthday;
                Edit5->Text=person.address;
                Edit6->Text=person.cardId;
                Edit7->Text=person.police;
                Edit8->Text=person.validStart;
                Edit9->Text=person.validEnd;
                Image1->Picture->LoadFromFile(TEXT("ssss.bmp"));
        }
        else
        {
                Edit1->Text=L"";
                Edit2->Text=L"";
                Edit3->Text=L"";
                Edit4->Text=L"";
                Edit5->Text=L"";
                Edit6->Text=L"";
                Edit7->Text=L"";
                Edit8->Text=L"";
                Edit9->Text=L"";
                Image1->Picture->Assign(NULL);
        }
        GetErrorText(szErrorText,32);
        EditResult->Text=szErrorText;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::BtnCloseClick(TObject *Sender)
{
        TCHAR szErrorText[32];

        CloseCardReader();
        GetErrorText(szErrorText,32);
        EditResult->Text=szErrorText;
}
//---------------------------------------------------------------------------

File: 普天二代证开发包2011/Delphi/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ExtCtrls;

type
  IDCARD_ALL = record
{    name : array[0..31] of char;
  sex : array[0..3] of char;
  nation : array[0..19] of char;
  birthday : array[0..11] of char;
  address : array[0..71] of char;
  cardId : array[0..19] of char;
  police: array[0..31] of char;
  validStart : array[0..11] of char;
  validEnd : array[0..11] of char;
  sexCode : array[0..3] of char;
  nationCode : array[0..3] of char;
  appendMsg : array[0..71] of char; }
  name:array[0..15]of WideChar;
  sex:array[0..1] of WideChar;
  nation:array[0..9] of WideChar;
  birthday:array[0..9] of WideChar;
  address:array[0..35] of WideChar;
  cardId: array[0..19] of WideChar;
  police:array[0..15] of WideChar;
  validStart:array[0..9] of WideChar;
  validEnd:array[0..9] of WideChar;
  sexCode:array[0..1] of WideChar;
  nationCode:array[0..3] of WideChar;
  appendMsg:array[0..35] of WideChar;
  end;
  TForm1 = class(TForm)
    Button2: TButton;
    Button1: TButton;
    Edit1: TEdit;
    Edit2: TEdit;
    Edit3: TEdit;
    Edit4: TEdit;
    Edit5: TEdit;
    Edit6: TEdit;
    Edit7: TEdit;
    Edit8: TEdit;
    Edit9: TEdit;
    Button3: TButton;
    Label1: TLabel;
    EditResult: TEdit;
    Label2: TLabel;
    Label3: TLabel;
    Label4: TLabel;
    Label5: TLabel;
    Label6: TLabel;
    Label7: TLabel;
    Label8: TLabel;
    Label9: TLabel;
    Image1: TImage;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure Button3Click(Sender: TObject);

  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;
  function OpenCardReader(lPort:LongInt; ulFlag:LongWord; ulBaudRate:LongWord): Longint; stdcall; external'cardapi3.dll';
  {function GetPersonMsgA(var card:IDCARD_ALL; pszImageFile:PChar): Longint; stdcall; external'cardapi3.dll';  }
  function GetPersonMsgW(var card:IDCARD_ALL; pszImageFile:PWideChar): Longint; stdcall; external 'cardapi3.dll';
  function CloseCardReader(): Longint;stdcall; external'cardapi3.dll';
  Procedure GetErrorTextW(pszBuffer:PWideChar; dwBufLen:LongWord); stdcall; external'cardapi3.dll';

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);
var
  errorText:array[0..31] of WideChar;
begin
  OpenCardReader(0,2,115200);
  GetErrorTextW(errorText,32);
  EditResult.Text:=errorText;
end;

procedure TForm1.Button2Click(Sender: TObject);
var
  rel: Integer;
  card: IDCARD_ALL;
  errorText:array[0..31] of WideChar;
begin
    rel := GetPersonMsgW(card,'ssss.bmp');
    if rel=0 then
    begin
      edit1.Text := card.name;
      edit2.Text := card.sex;
      edit3.Text := card.nation;
      edit4.Text := card.birthday;
      edit5.Text := card.address;
      edit6.Text := card.cardId;
      edit7.Text := card.police;
      edit8.Text := card.validStart;
      edit9.Text := card.validEnd;
      Image1.Picture.LoadFromFile('ssss.bmp');
    end
    else
    begin
      edit1.Text:='';
      edit2.Text:='';
      edit3.Text:='';
      edit4.Text:='';
      edit5.Text:='';
      edit6.Text:='';
      edit7.Text:='';
      edit8.Text:='';
      edit9.Text:='';
      Image1.Picture.Assign(nil);
    end;
  GetErrorTextW(errorText,32);
  EditResult.Text:=errorText;
end;

procedure TForm1.Button3Click(Sender: TObject);
var
  errorText:array[0..31] of WideChar;
begin
  CloseCardReader();
  GetErrorTextW(errorText,32);
  EditResult.Text:=errorText;
end;



end.
 